utils/bitfield/src/rleplus.rs

```diff
@@ -186,12 +183,7 @@ pub fn decode(enc: &BitVec<Lsb0, u8>) -> Result<BitVec<Lsb0, u8>, &'static str>
                     Some(true) => {
                         // Block Short
                         // prefix: 01
-                        let buf = enc
-                            .iter()
-                            .skip(i + 2)
-                            .take(4)
-                            .copied()
-                            .collect::<BitVec<Lsb0, u8>>();
+                        let buf = enc.iter().skip(i + 2).take(4).copied().collect::<BitVec>();
```

Comment:
Would it be overhead to generate this enc.iter().skip(i+2) at the top and then just add take() methods to it per the condition?

Reply:
Yup, that's fine, the none case shouldn't ever be hit. Making the change but I don't really see it as being more clear though.